File: 2 Introduction to Algorithms/9Sort/insert_sort/1_ insert_sort.py
# data: 1->n
def insert_sort(L):
	temp = L[0]
	for i in range(2, len(L)):	# range右边是开区间，与数组下标少1正好符合
		L[0] = L[i]
		j = i - 1
		while L[j] < L[0]:
			L[j+1] = L[j]
			j = j - 1
		L[j+1] = L[0]
	L[0] = temp

TEST = [0, 1, 4, 2, 7, 55, 32, 99]
print('Before Insert Sort:')
print(TEST)

insert_sort(TEST)

print('After Insert Sort:')
print(TEST)

File: 2 Data Structure and Data base/Chapter 2/Lab/1 insertsort.cpp
# include <iostream>

const int N = 9;

void InsertSort(int array[], int n);


int main()
{
	using std::cout;
	int A[N] = { 0, 2, 5, 3, 11, 5, 222, 77, 4 };
	
	cout << "Before insertion sort: \n";
	for (int i = 1; i < N; i++) {
		cout << A[i] << " ";
	}
	
	InsertSort(A, N);

	cout << "\nAfter insertion sort: \n";
	for (int i = 1; i < N; i++) {
		cout << A[i] << " ";
	}

	return 0;
}


void InsertSort(int array[], int n)
{
	int i;
	for (int j = 2; j < n; j++) {
		array[0] = array[j];
		i = j - 1;
		while (i > 0 && array[i] > array[0]) {
			array[i + 1] = array[i];
			i = i - 1;
		}
		array[i + 1] = array[0];
	}
}
